Packs/RadwareCloudDDoSProtectionServices/Integrations/RadwareCloudDDoSProtectionServices/RadwareCloudDDoSProtectionServices.py

```diff
@@ -102,86 +104,141 @@ def add_fields(events: list[dict], evnet_type: str | None):
+
+    Returns:
+        tuple: Contains the fetched documents and updated last run dictionary.
+    """
     end_time = int(time.time() * 1000)
-    start_time = last_run.get('last_fetch_alerts', end_time-2)+1
-    continue_fetch_alerts = last_run.get('continue_fetch_alerts', None)
+    last_fetch_key = f'last_fetch_{data_type}'
+    continue_fetch_key = f'continue_fetch_{data_type}'
+
+    latest_fetch = last_run.get(last_fetch_key, {})
+    start_time = latest_fetch.get('latest_timestamp', end_time - 1)
+    continue_fetch_data = last_run.get(continue_fetch_key, None)
     skip = 0
 
-    if continue_fetch_alerts:
-        end_time = continue_fetch_alerts.get('end_time')
-        start_time = continue_fetch_alerts.get('start_time')
-        skip = continue_fetch_alerts.get('fetched_alerts')
+    if continue_fetch_data:
+        end_time = continue_fetch_data.get('end_time')
+        start_time = continue_fetch_data.get('start_time')
+        skip = continue_fetch_data.get('fetched_' + data_type)
+
+    demisto.debug(f'RadwareCloudDDoS: {data_type=} {start_time=}, {end_time=}, {skip=}')
+
+    if data_type == 'events':
+        response = client.get_events(start_time, end_time, skip, PAGE_SIZE)
+    elif data_type == 'alerts':
+        response = client.get_alerts(start_time, end_time, skip, PAGE_SIZE)
+    else:
+        raise ValueError("Invalid data_type. Expected 'events' or 'alerts'.")
 
-    demisto.debug(f'RadwareCloudDDoS: {start_time=}, {end_time=},  {skip=}')
-    response = client.get_alerts(start_time, end_time, skip, 700)
     documents = response.get("documents")
-    demisto.debug(f'RadwareCloudDDoS: {len(documents)=}')
-    new_continue_fetch_alerts = {}
+    demisto.debug(f'RadwareCloudDDoS: {data_type=} {len(documents)=}')
+    new_continue_fetch_data = {}
 
-    if documents:
-        latest_timestamp_alerts = documents[0].get('context', {}).get("_timestamp")
-        demisto.debug(f'RadwareCloudDDoS: {latest_timestamp_alerts=}')
-        if len(documents) == 700:
-            demisto.debug('RadwareCloudDDoS: found next page')
-            new_continue_fetch_alerts = {'end_time': end_time, 'start_time': start_time, 'fetched_alerts': len(documents)+skip}
+    filtered_documents = filter_documents(documents, timestamp=latest_fetch.get('latest_timestamp'),
+                                          ids=latest_fetch.get('latest_ids'), data_type=data_type)
+    demisto.debug(f'RadwareCloudDDoS: {data_type=} {len(filtered_documents)=}')
 
-        if not continue_fetch_alerts:
-            last_run['last_fetch_alerts'] = latest_timestamp_alerts
-            demisto.debug(f'RadwareCloudDDoS: saved {latest_timestamp_alerts=}')
+    if filtered_documents:
 
-        if new_continue_fetch_alerts:
+        if len(filtered_documents) == PAGE_SIZE:
+            demisto.debug(f'RadwareCloudDDoS: {data_type=} found next page')
+            new_continue_fetch_data = {'end_time': end_time, 'start_time': start_time,
```

Comment:
Change continue_fetch_data and new_continue_fetch_data to iteration_cache and new_iteration_cache, respectively.